sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercionSuite.scala

```diff
@@ -348,6 +350,20 @@ class HiveTypeCoercionSuite extends PlanTest {
 
   test("type coercion for If") {
     val rule = HiveTypeCoercion.IfCoercion
+
+    // SPARK-14580 Hive IfCoercion should preserve predicate
+    case class AssertTrue(condition: Boolean) extends LeafExpression {
```

Comment:
maybe we should have this as an expression in spark sql, and register it so we don't need to delegate to hive.